ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp

```diff
@@ -34,39 +37,43 @@ Initializer::create_tainting_map()
 void
 Initializer::clear_tainting_map(const TaintRangeMapTypePtr& tx_map)
 {
-    if (not tx_map or tx_map->empty())
+    if (tx_map == nullptr) {
         return;
-
+    }
     if (const auto it = active_map_addreses.find(tx_map.get()); it == active_map_addreses.end()) {
-        // Map wasn't in the active addresses, do nothing
         return;
     }
 
     for (const auto& [fst, snd] : *tx_map) {
         snd.second->decref();
     }
-
     tx_map->clear();
```

Comment:
mutex here? Or at the start of the function (after lowering the one in clear_tainting_maps, see below)